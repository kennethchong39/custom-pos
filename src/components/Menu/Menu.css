.menu-title {
  text-align: center;
}

.menu-title h2 {
  margin: 0px;
}

.menu-items {
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
}

.item {
  display: inline-block;
  height: 40%;
  width: 45%;
  border: none;
  border-radius: 5%;
  background-color: #ff3a33;
  color: white;
  margin: 0.5rem;
  text-align: center;
  font-size: 1.2rem;
  padding: 1rem;
  cursor: pointer;
  outline: none;
}

.item button {
  border: none;
  outline: none;
  background-color: white;
  padding: 10px 10px;
  font-size: 0.8rem;
  width: 95%;
}

.item button:active {
  background-color: #d81913;
}
