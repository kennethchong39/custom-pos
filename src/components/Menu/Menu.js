import React from 'react';

import './Menu.css';

export default function Menu(props) {
  // to add an item to the cart
  const addToCart = (item) => {
    props.handleAdd(item);
  };

  return (
    <div className="menu">
      <div className="menu-title">
        <h2>Menu</h2>
      </div>
      <hr />
      <div className="menu-items">
        {props.content.map((item, index) => {
          return (
            <div className="item" key={index}>
              <h3>{item.name}</h3>
              <p>$ {item.price}</p>
              <button
                disabled={item.disabled}
                className={item.disabled ? 'disabled' : ''}
                onClick={() => addToCart({ ...item })}
              >
                {item.disabled ? 'OUT OF STOCK' : 'ADD TO CART'}
              </button>
            </div>
          );
        })}
      </div>
    </div>
  );
}
