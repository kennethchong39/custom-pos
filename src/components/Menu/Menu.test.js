import renderer from 'react-test-renderer';

import Menu from './Menu';

test('component: Menu tests', () => {
  const testProps = [
    {
      name: 'Vegetarian',
      price: 8.99,
      ingredients: {
        bread: 2,
        lettuce: 2,
        tomato: 2,
        cheese: 2,
      },
    },
    {
      name: 'BLT',
      price: 9.99,
      ingredients: {
        bread: 3,
        lettuce: 1,
        tomato: 1,
        bacon: 2,
      },
    },
    {
      name: 'Turkey',
      price: 10.99,
      ingredients: {
        bread: 2,
        lettuce: 1,
        tomato: 1,
        cheese: 1,
        turkey: 1,
      },
    },
  ];
  const tree = renderer.create(<Menu content={testProps} />).toJSON();

  expect(tree).toMatchSnapshot();
});
