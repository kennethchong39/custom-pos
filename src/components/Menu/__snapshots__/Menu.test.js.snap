// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Menu tests 1`] = `
<div
  className="menu"
>
  <div
    className="menu-title"
  >
    <h2>
      Menu
    </h2>
  </div>
  <hr />
  <div
    className="menu-items"
  >
    <div
      className="item"
    >
      <h3>
        Vegetarian
      </h3>
      <p>
        $ 
        8.99
      </p>
      <button
        className=""
        onClick={[Function]}
      >
        ADD TO CART
      </button>
    </div>
    <div
      className="item"
    >
      <h3>
        BLT
      </h3>
      <p>
        $ 
        9.99
      </p>
      <button
        className=""
        onClick={[Function]}
      >
        ADD TO CART
      </button>
    </div>
    <div
      className="item"
    >
      <h3>
        Turkey
      </h3>
      <p>
        $ 
        10.99
      </p>
      <button
        className=""
        onClick={[Function]}
      >
        ADD TO CART
      </button>
    </div>
  </div>
</div>
`;
